src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.CrossGen.targets

```diff
@@ -0,0 +1,231 @@
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="coreclr" />
+
+    <NETSdkError Condition="'@(_JitResolvedPath->Count())' &gt; 1"
+    ResourceName="MultipleFilesResolved"
+    FormatArguments="jit" />
+
+    <NETSdkError Condition="'@(_JitResolvedPath)'== ''"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="jit" />
+    
+    
+    
+    <!-- Get the crossgen and jit path-->
+    <PropertyGroup>
+      <_CoreclrPath>@(_CoreclrResolvedPath)</_CoreclrPath>
+      <_JitPath>@(_JitResolvedPath)</_JitPath>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrPath)))</_CoreclrDir>
+      <_CoreclrPkgDir>$([System.IO.Path]::Combine($(_CoreclrDir),"..\..\..\"))</_CoreclrPkgDir>
+      <_CrossgenDir>$([System.IO.Path]::Combine($(_CoreclrPkgDir),"tools"))</_CrossgenDir>
+      <!-- TODO override with rid specific tools path for x-arch -->
+      <_Crossgen>$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen"))</_Crossgen>
+      <_Crossgen Condition="$(OS.ToUpper().Contains('WINDOWS'))">$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen.exe"))</_Crossgen>
+    </PropertyGroup>
+
+    <NETSdkError Condition="!Exists($(_Crossgen))"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="$(_Crossgen)" />
+
+    <PropertyGroup>
+      <_Crossgen> $([System.IO.Path]::GetFullPath($(_Crossgen)))</_Crossgen>
+    </PropertyGroup>
+
+    <!-- Create the output directory where crossgened assemblies are going to be present -->
+    <ItemGroup>
+      <_RuntimeOptimizedDirectories Include="@(_AssembliestoCrossGen -> '$(_RuntimeOptimizedDir)\%(RecursiveDir)')"/>
+    </ItemGroup>
+    <MakeDir  Directories="@(_RuntimeOptimizedDirectories)"/>
+
+    <!-- CrossGen the assemblies  -->
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="RunCrossGen"
+                 Properties="_CGenExe=$(_Crossgen);
+                             _CGenJit=$(_JitPath);
+                             _CGenInputAssembly=%(_AssembliestoCrossGen.Fullpath);
+                             _CGenOutputAssembly=$([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension)));
+                             _CGenPlatformAssembliesPath=$(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir);" />
+
+
+    <!-- Copy by proxy the directory structure to PublishDir-->
+    <ItemGroup>
+      <_ManagedResolvedFileToPublish Include="$(_RuntimeOptimizedDir)\**\*.*" />
+    </ItemGroup>
+  </Target>
+
+  <!--
+    ============================================================
+                                        RunCrossGen
+    Target Encapsulating the crossgen command  
+    ============================================================
+    -->
+  <Target Name="RunCrossGen">
+    <Message Text="$(_CGenExe) -readytorun -in $(_CGenInputAssembly) -out $(_CGenOutputAssembly) -jitpath $(_CGenJit) -platform_assemblies_paths $(_CGenPlatformAssembliesPath)"/>
```

Comment:
This is only used by dotnet cache, which only takes a list of packages. This is not to be confused with build of publish.

Having said that, it is to plumb this logic by appending to _CGenPlatformAssembliesPath as  done at  L95:
  _CGenPlatformAssembliesPath=$(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir);" />